roachprod: stage GEOS libraries to clusters

otan opened the pull request:
This commit adds the libgeos* and libgeos_c* files to be staged on
roachprod. If the files could not be found, they are gracefully handled
in both the library and release cases.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/51386)
<!-- Reviewable:end -->

petermattis reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/51386)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan and @petermattis)

___
*[pkg/cmd/roachprod/main.go, line 1317 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/51386#-MCgyfai1g6gGlVqcaF4:-MCgyfai1g6gGlVqcaF5:b1rual) ([raw file](https://github.com/cockroachdb/cockroach/blob/1da50995014d7967e4fdd4cf412f84454b2f4f68/pkg/cmd/roachprod/main.go#L1317)):*
> ```Go
> 					library,
> 					fmt.Sprintf("cockroach/lib/%s", library),
> 					versionArg,
> ```

If `versionArg` is empty, then we'll be finding the latest SHA multiple times and could be resolving it differently. Not sure if that is a problem worth fixing. Is it possible for `StageRemoteBinary` to return the SHA it staged?
___
*[pkg/cmd/roachprod/install/staging.go, line 84 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/51386#-MCgy2oFDNILhROgc4ib:-MCgy2oFDNILhROgc4ic:b-6iw9zf) ([raw file](https://github.com/cockroachdb/cockroach/blob/1da50995014d7967e4fdd4cf412f84454b2f4f68/pkg/cmd/roachprod/install/staging.go#L84)):*
> ```Go
> 	fmt.Printf("Resolved library url for %s: %s\n", libraryName, url)
> 	cmdStr := fmt.Sprintf(
> 		`sudo mkdir -p /usr/local/lib/cockroach && \
> ```

The staged cockroach binary is installed into `$HOME`. For consistency I think we should put the geos libraries in `$HOME/lib`


<!-- Sent from Reviewable.io -->

otan committed:
roachprod: stage GEOS libraries to clusters

This commit adds the libgeos* and libgeos_c* files to be staged on
roachprod. If the files could not be found, they are gracefully handled
in both the library and release cases.

Release note: None

otan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/51386)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @petermattis)

___
*[pkg/cmd/roachprod/main.go, line 1317 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/51386#-MCgyfai1g6gGlVqcaF4:-MCh79LKD-uoD2JMgaxV:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/1da50995014d7967e4fdd4cf412f84454b2f4f68/pkg/cmd/roachprod/main.go#L1317)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

If `versionArg` is empty, then we'll be finding the latest SHA multiple times and could be resolving it differently. Not sure if that is a problem worth fixing. Is it possible for `StageRemoteBinary` to return the SHA it staged?
</blockquote></details>

Done.
___
*[pkg/cmd/roachprod/install/staging.go, line 84 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/51386#-MCgy2oFDNILhROgc4ib:-MCh0N1g9XmIpl8VX_IZ:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/1da50995014d7967e4fdd4cf412f84454b2f4f68/pkg/cmd/roachprod/install/staging.go#L84)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

The staged cockroach binary is installed into `$HOME`. For consistency I think we should put the geos libraries in `$HOME/lib`
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

petermattis reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/51386#-:-MChY1zy08cqccWbsNSc:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
tftr!

bors r=petermattis
